modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -4781,18 +4781,32 @@ void GDScriptAnalyzer::validate_call_arg(const List<GDScriptParser::DataType> &p
 
 		if ((arg_type.is_variant() || !arg_type.is_hard_type()) && !(par_type.is_hard_type() && par_type.is_variant())) {
 			// Argument can be anything, so this is unsafe.
+#ifdef DEBUG_ENABLED
 			mark_node_unsafe(p_call->arguments[i]);
+			String visible_arg_type = arg_type.to_string();
+			if (!arg_type.is_hard_type()) {
+				visible_arg_type = "Variant";
+			}
+			parser->push_warning(p_call->arguments[i], GDScriptWarning::UNSAFE_CALL_ARGUMENT, itos(i + 1), p_call->function_name, par_type.to_string(), visible_arg_type);
+#endif
 		} else if (par_type.is_hard_type() && !is_type_compatible(par_type, arg_type, true)) {
 			// Supertypes are acceptable for dynamic compliance, but it's unsafe.
+#ifdef DEBUG_ENABLED
 			mark_node_unsafe(p_call);
+			String visible_arg_type = arg_type.to_string();
+			if (!arg_type.is_hard_type()) {
+				visible_arg_type = "Variant";
+			}
+			parser->push_warning(p_call->arguments[i], GDScriptWarning::UNSAFE_CALL_ARGUMENT, itos(i + 1), p_call->function_name, par_type.to_string(), visible_arg_type);
+#endif
```

Comment:
I think this should be in the `else` branch of the `if` just below. There shouldn't be an error and a warning reported for the same thing at the same time.